c/cert/src/rules/CON34-C/AppropriateThreadObjectStorageDurations.ql

```diff
@@ -0,0 +1,46 @@
+/**
+ * @id c/cert/appropriate-thread-object-storage-durations
+ * @name CON34-C: Declare objects shared between threads with appropriate storage durations
+ * @description Accessing thread-local variables with automatic storage durations can lead to
+ *              unpredictable program behavior.
+ * @kind problem
+ * @precision high
+ * @problem.severity error
+ * @tags external/cert/id/con34-c
+ *       correctness
+ *       concurrency
+ *       external/cert/obligation/rule
+ */
+
+import cpp
+import codingstandards.c.cert
+import codingstandards.cpp.Concurrency
+import semmle.code.cpp.dataflow.TaintTracking
+import semmle.code.cpp.dataflow.DataFlow
+
+class MallocFunctionCall extends FunctionCall {
+  MallocFunctionCall() { getTarget().getName() = "malloc" }
+}
+
+from C11ThreadCreateCall tcc, StackVariable sv, Expr arg, Expr acc
+where
+  not isExcluded(tcc, Concurrency4Package::appropriateThreadObjectStorageDurationsQuery()) and
+  tcc.getArgument(2) = arg and
+  sv.getAnAccess() = acc and
+  // a stack variable that is given as an argument to a thread
+  TaintTracking::localTaint(DataFlow::exprNode(acc), DataFlow::exprNode(arg)) and
+  // it's either not static
+  not sv.isStatic() and
```

Comment:
and I think this is redundant to the implementation of StackVariable also

Reply:
Good catch, I think it is redundant to say both.